### PR TITLE
Make OAuth2User extends OAuth2AuthenticatedPrincipal

### PR DESCRIPTION
**Pull request by eddumelendez**
Fixes gh-7378

### CODE REVIEW

File: oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/user/OAuth2User.java [hunk 1]
```diff
@@ -55,11 +55,4 @@
 	 */
 	Collection<? extends GrantedAuthority> getAuthorities();
```
Comment (jzheaux) [inline oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/user/OAuth2User.java]: It's not clear to me why this method would remain. I think `OAuth2AuthenticatedPrincipal` has a `getAuthorities` as well?

Comment (eddumelendez) [reply oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/user/OAuth2User.java]: Yes, I missed that one. PR has ben updated

File: oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/userinfo/CustomUserTypesOAuth2UserServiceTests.java [hunk 2]
```diff
@@ -134,10 +135,10 @@ public void loadUserWhenUserInfoSuccessResponseThenReturnUser() {
 
 		assertThat(user.getName()).isEqualTo("first last");
 		assertThat(user.getAttributes().size()).isEqualTo(4);
-		assertThat(user.getAttributes().get("id")).isEqualTo("12345");
-		assertThat(user.getAttributes().get("name")).isEqualTo("first last");
-		assertThat(user.getAttributes().get("login")).isEqualTo("user1");
-		assertThat(user.getAttributes().get("email")).isEqualTo("user1@example.com");
+		assertThat((Object) user.getAttribute("id")).isEqualTo("12345");
```
Comment (jgrandja) [inline oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/userinfo/CustomUserTypesOAuth2UserServiceTests.java]: Change `Object` to `String`

File: oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/userinfo/DefaultOAuth2UserServiceTests.java [hunk 3]
```diff
@@ -146,12 +147,12 @@ public void loadUserWhenUserInfoSuccessResponseThenReturnUser() {
 
 		assertThat(user.getName()).isEqualTo("user1");
 		assertThat(user.getAttributes().size()).isEqualTo(6);
-		assertThat(user.getAttributes().get("user-name")).isEqualTo("user1");
-		assertThat(user.getAttributes().get("first-name")).isEqualTo("first");
-		assertThat(user.getAttributes().get("last-name")).isEqualTo("last");
-		assertThat(user.getAttributes().get("middle-name")).isEqualTo("middle");
-		assertThat(user.getAttributes().get("address")).isEqualTo("address");
-		assertThat(user.getAttributes().get("email")).isEqualTo("user1@example.com");
+		assertThat((Object) user.getAttribute("user-name")).isEqualTo("user1");
```
Comment (jgrandja) [inline oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/userinfo/DefaultOAuth2UserServiceTests.java]: Change `Object` to `String`

File: oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/userinfo/DefaultReactiveOAuth2UserServiceTests.java [hunk 4]
```diff
@@ -137,12 +138,12 @@ public void loadUserWhenUserInfoSuccessResponseThenReturnUser() throws Exception
 
 		assertThat(user.getName()).isEqualTo("user1");
 		assertThat(user.getAttributes().size()).isEqualTo(6);
-		assertThat(user.getAttributes().get("id")).isEqualTo("user1");
-		assertThat(user.getAttributes().get("first-name")).isEqualTo("first");
-		assertThat(user.getAttributes().get("last-name")).isEqualTo("last");
-		assertThat(user.getAttributes().get("middle-name")).isEqualTo("middle");
-		assertThat(user.getAttributes().get("address")).isEqualTo("address");
-		assertThat(user.getAttributes().get("email")).isEqualTo("user1@example.com");
+		assertThat((Object) user.getAttribute("id")).isEqualTo("user1");
```
Comment (jgrandja) [inline oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/userinfo/DefaultReactiveOAuth2UserServiceTests.java]: Change `Object` to `String`

**Comment by eddumelendez** [comment]
Thanks @jgrandja ! I have updated the PR

**Comment by jzheaux** [comment]
Thanks, @eddumelendez, this is now merged into `master`.

**Pull request merged**